Userland/Libraries/LibWeb/Painting/StackingContext.cpp

```diff
@@ -100,6 +100,33 @@ void StackingContext::paint_svg(PaintContext& context, PaintableBox const& paint
     if (phase != PaintPhase::Foreground)
         return;
 
+    auto opacity = paintable.computed_values().opacity();
+    if (opacity == 0.0f)
+        return;
+
+    DisplayListRecorderStateSaver saver(context.display_list_recorder());
+
+    auto to_device_pixels_scale = float(context.device_pixels_per_css_pixel());
+    Gfx::IntRect source_paintable_rect = context.enclosing_device_rect(paintable.absolute_paint_rect()).to_type<int>();
+
+    auto transform_matrix = paintable.transform();
+    Gfx::FloatPoint transform_origin = paintable.transform_origin().to_type<float>();
+
+    DisplayListRecorder::PushStackingContextParams push_stacking_context_params {
+        .opacity = opacity,
+        .filter = paintable.computed_values().filter(),
+        .is_fixed_position = paintable.is_fixed_position(),
+        .source_paintable_rect = source_paintable_rect,
+        .transform = {
+            .origin = transform_origin.scaled(to_device_pixels_scale),
+            .matrix = transform_matrix,
+            // FIXME: this is resulting in a compile error in which this method
+            // is not being found
+            // .matrix = matrix_with_scaled_translation(transform_matrix, to_device_pixels_scale),
+        },
+    };
+    context.display_list_recorder().push_stacking_context(push_stacking_context_params);
```

Comment:
we used PushStackingContext for SVG until recently, but it was a mistake because stacking context concept does not exist in SVG painting model. see https://github.com/LadybirdBrowser/ladybird/pull/1697

the correct way to implement support for filters in SVG would be to introduce a new display list item specifically responsible for the application of filters. see ApplyMaskBitmap, ApplyTransform, and ApplyOpacity as examples.

Reply:
@kalenikaliaksandr are you advocating for a new set of display list items, or just one new item that applies an ordered list of CSS filters such as the logic we currently have for stacking contexts, and then does so for both stacking contexts and SVG filters?

Reply:
I would prefer to move filters application logic from PushStackingContext into new ApplyFilters command and use it for both CSS and SVG.

Reply:
I looked at the ApplyOpacity, ApplyMaskBitmap, and ApplyTransform commands, and it seems like the ApplyFilters can be made the same way, by using `ImmutableComputedValues::ResolvedFilter`'s data. We can use `push_stacking_context` in `Userland/Libraries/LibWeb/Painting/DisplayListPlayerSkia.cpp` as a reference for the inner implementation.

Once this is done, would we have to call the new `DisplayListRecorder::apply_filters` method inside SVGSVGPaintable as well as in PaintableBox's implementation of paint/paint_descendants?